criu/sk-inet.c

```diff
@@ -875,6 +875,8 @@ static int open_inet_sk(struct file_desc *d, int *new_fd)
 		goto err;
 	if (restore_opt(sk, SOL_SOCKET, SO_REUSEPORT, &yes))
 		goto err;
+	if (restore_bound_opts(sk, ie->opts))
```

Comment:
pls you add a comment that explains why restore_bound_opts has to be called before connect

Reply:
@avagin 
My runing environmet configuration:
> #ip route show table 40

> 127.127.127.0/28 dev eth0 proto kernel scope link src 127.127.127.10

> route 
> Kernel IP routing table
> Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
> default         8.44.32.1       0.0.0.0         UG    0      0        0 eth4

when restore, kernel find the output interface of eth4. 

`@avagin 
My runing environmet configuration:
> #ip route show table 40

> 127.127.127.0/28 dev eth0 proto kernel scope link src 127.127.127.10

> route 
> Kernel IP routing table
> Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
> default         8.44.32.1       0.0.0.0         UG    0      0        0 eth4

when restore, kernel find the output interface of eth4, then calling __mkroute_output method to add the route table,
it will return -22 from here.

`        if (likely(!IN_DEV_ROUTE_LOCALNET(in_dev)))
        {
                if (ipv4_is_loopback(fl4->saddr) &&
                    !(dev_out->flags & IFF_LOOPBACK) &&
                    !netif_is_l3_master(dev_out))
                {
                        return ERR_PTR(-EINVAL);
                }
        }
`

So I thidk if we restore the bound device before connect, it will find the output device of eth0.